cli/src/stake.rs

```diff
@@ -551,18 +566,37 @@ pub fn process_create_stake_account(
         withdrawer: withdrawer.unwrap_or(config.keypair.pubkey()),
     };
 
-    let ixs = stake_instruction::create_account(
-        &config.keypair.pubkey(),
-        &stake_account_pubkey,
-        &authorized,
-        lockup,
-        lamports,
-    );
+    let ixs = if let Some(seed) = seed {
+        stake_instruction::create_account_with_seed(
+            &config.keypair.pubkey(), // from
+            &stake_account_address,   // to
+            &stake_account_pubkey,    // base
+            seed,                     // seed
+            &authorized,
+            lockup,
+            lamports,
+        )
+    } else {
+        stake_instruction::create_account(
+            &config.keypair.pubkey(),
+            &stake_account_pubkey,
+            &authorized,
+            lockup,
+            lamports,
+        )
+    };
     let (recent_blockhash, fee_calculator) = rpc_client.get_recent_blockhash()?;
+
+    let signers = if stake_account_pubkey != config.keypair.pubkey() {
```

Comment:
@jackcmay, here's a nice example of the pain that comes from passing `&mut Account` instead of `RefCell<Account>` from the runtime to programs.

Reply:
how's this related, @garious ?

Reply:
If we passed RefCell, then we can remove the assertion in Message that prevents you from passing the same key twice here.  Both RefCells would point to the same Account and it would be up to the program implementation not to access them at the same time, which is easy and safe to do.

Reply:
the guard is actually there because transaction signing code is brittle...

Reply:
Brittle how? There's an intentional assertion to guard against adding duplicate public keys. Is it overly conservative?

Reply:
building a transaction then signing a transaction currently requires that the caller knows which keys and what order they appear in the instruction vector.  this is maybe a shortcoming of our _instruction.rs files